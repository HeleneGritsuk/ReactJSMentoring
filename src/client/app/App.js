import React from "react";
import { hot } from "react-hot-loader";
import { Provider } from "react-redux";
import { Route, Switch, Redirect } from "react-router-dom";
import SearchScreen from "./Screens/SearchPage";
import FilmScreen from "./Screens/FilmScreen";
import NotFoundScreen from "./Screens/404";
import "./style.css";

const App = ({ Router, location, context, store }) => (
  <Provider store={store}>
    <Router location={location} context={context}>
      <Switch>
        <Route exact path="/" component={SearchScreen} />
        <Route path="/search/:searchQuery" component={SearchScreen} />
        <Route path="/film/:id" component={FilmScreen} />
        <Route path="*" component={NotFoundScreen} />
      </Switch>
    </Router>
  </Provider>
);

export default hot(module)(App);
