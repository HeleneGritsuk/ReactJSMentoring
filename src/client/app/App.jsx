import React from 'react';
import { render } from 'react-dom';
import SearchScreen from './Screens/SearchPage';
import FilmScreen from './Screens/FilmScreen';
import ErrorBoundary from './components/ErrorBoundary';
import './style.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showSearchPage: false,
      showFilmPage: false
    }
    this.handleSearchPageClick = this.handleSearchPageClick.bind(this);
    this.handleFilmPageClick = this.handleFilmPageClick.bind(this);
    this.throwError = this.throwError.bind(this);
  }
  handleSearchPageClick() {
    this.setState({
      showSearchPage: true,
      showFilmPage: false
     });
  }
  handleFilmPageClick() {
    this.setState({
      showSearchPage: false,
      showFilmPage: true
     });
  }
  throwError() {
    throw new Error('I crashed!');
  }

  render () {
    return(
      <ErrorBoundary>
        <div className='main-wrapper'>
          <div className = 'screensBtnWrapper'>
            <button onClick = {this.handleSearchPageClick}>Show SearchPage view</button>
            <button onClick = {this.handleFilmPageClick}>Show FilmPage view</button>
            <button onClick = {this.throwError}> Throw Error</button>
          </div>
          {this.state.showSearchPage ? (<SearchScreen/>) : null}
          {this.state.showFilmPage ? (<FilmScreen goToHomePage = {this.handleSearchPageClick}/>) : null}
        </div>
      </ErrorBoundary>
    );
  }
}

export default App;
