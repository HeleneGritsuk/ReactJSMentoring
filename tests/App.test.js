import React from 'react';
import App from '../src/client/app/App';
import renderer from 'react-test-renderer';

describe('App', () => {
  it('should render App correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
