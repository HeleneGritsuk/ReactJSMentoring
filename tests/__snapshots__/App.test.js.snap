// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App correctly 1`] = `
<div>
  <h1
    className="title"
  >
    React core concepts task
  </h1>
  <div>
    <p>
      React components can be created in different ways, using:
    </p>
    <li>
      React.createElement
    </li>
    <li>
      React.Component
    </li>
    <li>
      React.PureComponent
    </li>
    <li>
      functional components
    </li>
  </div>
  <footer>
     React mentoring program 2018 
  </footer>
</div>
`;
